x/evm/keeper/state_transition.go

```diff
@@ -326,11 +327,13 @@ func (k *Keeper) ApplyMessageWithConfig(ctx sdk.Context,
 
 	// set the custom precompiles to the EVM (if any)
 	if cfg.Params.HasCustomPrecompiles() {
```

Comment:
Let's create a function for the inner contents of this method to reduce complexity and make it more maintainable

Reply:
Great suggestion! wanted just to prove this works and fix the bug in the original PR to unblock tests but will clean this one tomorrow and implement your suggestions! Will move it to draft.